### PR TITLE
AC-1863 - (UI) Create a global <Form/> component and configure existing form fields to be easily trackable

### PR DESCRIPTION
**Pull request by shrirupa**
AC-1863

### What Changed

- Create a global `<Form>` component.
- Replace all `<form>` with <Form>
- added id to all `<Form>`
- added eslint rule to throw a waning when `<form>` is used
- Update lot of snapshots 😅 

### How To Test

- Check no active pages have html `<form>` and are replaced by `<Form>` with ids

### To Do

- [ ] Address feedback

### CODE REVIEW

File: src/pages/support/PremiumSupportPage.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ import { createTicket } from 'src/actions/support';
 import { showAlert } from 'src/actions/globalAlert';
 import { PageLink } from 'src/components/links';
 import { Button, Page, Panel } from 'src/components/matchbox';
+import { Form } from 'src/components/form';
 import PremiumSupportFields from './components/PremiumSupportFields';
 import { generateMessage } from './helpers/formHelpers';
```
Comment (shrirupa) [inline src/pages/support/PremiumSupportPage.js:11]: Check with accounts team if we still have aws customers ?

**Comment by PatrickSison** [comment]
There's still a form tag on `src/pages/trackingDomains/components/CreateForm.js`

**Comment by PatrickSison** [comment]
There's also another on `src/pages/sendingDomains/components/AssignTrackingDomain.js`

**Comment by PatrickSison** [comment]
I think a couple of small places, but otherwise, good. Seems like it still is working fine

File: src/pages/templates/CreatePage.js [hunk 2]
```diff
@@ -53,7 +54,7 @@ export default class CreatePage extends Component {
 
     return (
       <Page breadcrumbAction={backAction} title="Create Template">
-        <form onSubmit={handleSubmit(this.handleCreate)}>
+        <Form onSubmit={handleSubmit(this.handleCreate)}>
```
Comment (nstanard) [inline src/pages/templates/CreatePage.js:57]: Does this need an `id`?

**Comment by nstanard** [comment]
Do you know if any of the references updated have multiple instances displayed on a single page?

**Comment by shrirupa** [comment]
@PatrickSison  for https://github.com/SparkPost/2web2ui/pull/2080#issuecomment-788078974 and https://github.com/SparkPost/2web2ui/pull/2080#issuecomment-788083393 these are actually OG pages and don't show up in app anymore, that is why left them out. In fact, I have removed all src/pages/trackingDomains here - https://github.com/SparkPost/2web2ui/pull/2094

File: src/pages/templates/CreatePage.js (hunk 2 above)
Comment (shrirupa) [reply src/pages/templates/CreatePage.js:57]: ahh I missed it! Thanks!

**Comment by shrirupa** [comment]
> Do you know if any of the references updated have multiple instances displayed on a single page?

@nstanard That's a good thing to check. I am not aware of any such instance where duplicate forms show up on the same page, but I will check.

File: src/pages/support/PremiumSupportPage.js (hunk 1 above)
Comment (shrirupa) [reply src/pages/support/PremiumSupportPage.js:11]: still supporting them

File: .eslintrc [hunk 3]
```diff
@@ -32,6 +32,7 @@
     "lodash/prefer-some": "off",
     "lodash/prefer-startswith": "off",
     "lodash/prefer-times": "off",
+    "react/forbid-elements": ["warn", { "forbid": [{ "element": "form", "message": "Please use the global <Form /> component in src/components/form, which helps in global tracking" }] }],
```
Comment (nstanard) [inline .eslintrc:35]: I want to propose we turn this into a custom rule and put it in `.eslintplugins`

I quickly threw together a rule that I think will work for us:

`require-form-component-with-id.js`
```
const requireFormComponentWithId = {
  meta: {},
  create(context) {
    const { report } = context;
    const isTestFile = /.test.js$/.test(context.getFilename());

    if (isTestFile) {
      return;
    }

    return {
      JSXElement: (node) => {
        if (node.openingElement.name.name === 'form') {
          report({
            node: node,
            message: 'Please use the global <Form /> component in src/components/form, which helps in global tracking.',
          });
        }

        if (node.openingElement.name.name === 'Form') {
          const idAttribute = node.openingElement.attributes.find(attr => {
            if (attr && attr.name && attr.name.name && attr.name.name === 'id') {
              return attr;
            }
          });

          if (!idAttribute) {
            report({
              node: node,
              message: 'Please use an id="" attribute with the global <Form /> component.',
            });
          }
        }
      }
    };
  },
};

module.exports = requireFormComponentWithId;
```

`require-form-component-with-id.test.js`
```
import runner from '../utils/eslintTestRunner';
import rule from '../require-form-component-with-id';

runner('require-form-component-with-id', rule, {
  valid: {
    'Form instance with an id': {
      code: `<Form id="valid"></Form>`
    }
  },
  invalid: {
    'normal html form instance': {
      code: `<form></form>`,
      errors: [{ message: 'Please use the global <Form /> component in src/components/form, which helps in global tracking.' }]
    },
    'global Form component instance without an id': {
      code: `<Form></Form>`,
      errors: [{ message: 'Please use an id="" attribute with the global <Form /> component.' }]
    }
  },
});
```

Comment (shrirupa) [reply .eslintrc:35]: talked about this in slack.

**Comment by shrirupa** [comment]
@PatrickSison @nstanard  This PR is ready for another pass, whenever you have some time.

File: src/components/form/Form.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+/* eslint-disable react/forbid-elements */
+import React from 'react';
+import PropTypes from 'prop-types';
+
+function handleSubmit() {}
+
+document.addEventListener('submit', handleSubmit, false);
```
[the remark this answers is not in the document]
Comment (shrirupa) [reply src/components/form/Form.js:7]: nope! we will handle that in future PR.

**Pull request merged**